src/components/OrderBookWidget.tsx

```diff
@@ -224,103 +223,257 @@ function _printOrderBook(pricePoints: PricePointDetails[], baseToken: TokenDetai
+    // To allow we to zoom reaaaaly tiny fractions
+    xAxis.maxZoomFactor = 10 ** 18
+
+    const yAxis = _chart.yAxes.push(new am4charts.ValueAxis())
+
+    yAxis.keepSelection = true
+    yAxis.maxZoomFactor = 10 ** 18
+
+    // Colors
+    const colors = {
+      green: '#3d7542',
+      red: '#dc1235',
+    }
+
+    // Create series
+    const bidSeries = _chart.series.push(new am4charts.StepLineSeries())
+    bidSeries.dataFields.valueX = 'priceNumber'
+    bidSeries.dataFields.valueY = 'bidValueY'
+    bidSeries.strokeWidth = 1
+    bidSeries.stroke = am4core.color(colors.green)
+    bidSeries.fill = bidSeries.stroke
+    bidSeries.fillOpacity = 0.1
+
+    const askSeries = _chart.series.push(new am4charts.StepLineSeries())
+    askSeries.dataFields.valueX = 'priceNumber'
+    askSeries.dataFields.valueY = 'askValueY'
+    askSeries.strokeWidth = 1
+    askSeries.stroke = am4core.color(colors.red)
+    askSeries.fill = askSeries.stroke
+    askSeries.fillOpacity = 0.1
+
+    // Add cursor
+    _chart.cursor = new am4charts.XYCursor()
+
+    _chart.cursor.snapToSeries = [bidSeries, askSeries]
+
+    am4core.useTheme(am4themesSpiritedaway)
+    am4core.options.autoSetClassName = true
+
+    _chart.scrollbarX = new am4core.Scrollbar()
+    _chart.scrollbarY = new am4core.Scrollbar()
 
-export default OrderBookWidget
+    setChart(_chart)
+
+    return (): void => _chart.dispose()
+    // We'll create only one instance as long as the component is not unmounted
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (chart) {
+      const baseTokenLabel = safeTokenName(baseToken)
+      const quoteTokenLabel = safeTokenName(quoteToken)
+      const market = baseTokenLabel + '-' + quoteTokenLabel
+
+      const networkDescription = networkId !== Network.Mainnet ? `${getNetworkFromId(networkId)} ` : ''
+
+      // Axes configs
+      const xAxis = chart.xAxes.values[0] as am4charts.ValueAxis<am4charts.AxisRenderer>
+      xAxis.title.text = `${networkDescription} Price (${quoteTokenLabel})`
+
+      const yAxis = chart.yAxes.values[0] as am4charts.ValueAxis<am4charts.AxisRenderer>
+      yAxis.title.text = baseTokenLabel
+
+      // Add data
+      chart.dataSource.url = dexPriceEstimatorApi.getOrderBookUrl({
+        baseTokenId: baseToken.id,
+        quoteTokenId: quoteToken.id,
+        hops,
+        networkId,
+      })
+
+      // Removing any previous event handler
+      chart.dataSource.adapter.remove('parsedData')
+      // Adding new event handler
+      chart.dataSource.adapter.add('parsedData', data => {
+        try {
+          const bids = processData(data.bids, baseToken, quoteToken, Offer.Bid)
+          const asks = processData(data.asks, baseToken, quoteToken, Offer.Ask)
+          const pricePoints = bids.concat(asks)
+
+          let startX = 0
+          let endX = 1
+
+          if (bids.length > 0 && asks.length > 0) {
+            const minBid = bids[0].priceNumber
```

Comment:
Probably you don't want to deal with this in the PR, but we can ad a TODO here with an issue to select a maxBid/minAsk taking the quote price into account.

If we have the OWL or USD price for the quote, we can:
* Filter tiny orders: i.e < 5-10 OWL (it would be great to align this with the warning message for the minimal trade, or at least make this tiny orders filter be a subset of the trades that would warn if you try to send them)
* Select as the minAsk/maxBid the price where we accumulate at least 30 or 50 OWL

Reply:
> Filter tiny orders

Yes, this is what I intend to do in this PR to resolve the issue with a fixed filter amount of 0.01, which doesn't work when dealing with WETH, creating asymmetrical graphs when inverting the token order.

> Select as the minAsk/maxBid the price where we accumulate at least 30 or 50 OWL
In that case we might have no orderbook under some circumstances.